scene/resources/material.cpp

```diff
@@ -684,29 +684,35 @@ void BaseMaterial3D::_update_shader() {
 		return; //no update required in the end
 	}
 
-	MutexLock lock(shader_map_mutex);
-	if (shader_map.has(current_key)) {
-		shader_map[current_key].users--;
-		if (shader_map[current_key].users == 0) {
-			// Deallocate shader which is no longer in use.
-			RS::get_singleton()->free(shader_map[current_key].shader);
-			shader_map.erase(current_key);
+	{
+		MutexLock lock(shader_map_mutex);
+		if (ShaderData *v = shader_map.getptr(current_key); v) {
+			v->users--;
+			if (v->users == 0) {
+				// Deallocate shader which is no longer in use.
+				RS::get_singleton()->free(v->shader);
```

Comment:
An improvement, as we no longer perform multiple lookups into `shader_map` too

Reply:
This is a great pattern that is worth using more widely. Calling `has()` is almost always a waste of cycles